src/network/networkd-ndisc.c

```diff
@@ -380,15 +379,42 @@ static int ndisc_router_process_icmp6_ratelimit(Link *link, sd_ndisc_router *rt)
 
         /* Limit the maximal rates for sending ICMPv6 packets. 0 to disable any limiting, otherwise the
          * minimal space between responses in milliseconds. Default: 1000. */
-        xsprintf(buf, USEC_FMT, DIV_ROUND_UP(icmp6_ratelimit, USEC_PER_MSEC));
-
-        r = sysctl_write_ip_property(AF_INET6, NULL, "icmp/ratelimit", buf);
+        r = sysctl_write_ip_property_uint32(AF_INET6, NULL, "icmp/ratelimit",
+                (uint32_t)DIV_ROUND_UP(icmp6_ratelimit, USEC_PER_MSEC));
         if (r < 0)
                 log_link_warning_errno(link, r, "Failed to apply ICMP6 ratelimit, ignoring: %m");
 
         return 0;
 }
 
+static int ndisc_router_process_retransmission_time(Link *link, sd_ndisc_router *rt) {
+        usec_t retrans_time;
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(rt);
+
+        if (!link->network->ipv6_accept_ra_use_retransmission_time)
+                return 0;
+
+        r = sd_ndisc_router_get_retransmission_time(rt, &retrans_time);
+        if (r < 0) {
+                log_link_debug(link, "Failed to get ICMP6 ratelimit from RA, ignoring: %m");
+                return 0;
+        }
+
+        /* Set the retransmission time for Neigbor Solicitations.
+         * 0 is the unspecified value and must not be set (see RFC4861, 6.3.4) */
+        if (timestamp_is_set(retrans_time)) {
+                r = sysctl_write_ip_neighbor_property_uint32(AF_INET6, link->ifname, "retrans_time_ms",
+                        (uint32_t)DIV_ROUND_UP(retrans_time, USEC_PER_MSEC));
```

Comment:
The value is given by originally msec, and converted to usec. So this should be safe. But, for extra safety, please do the same way as #30898 ? That is, check the range of the `DIV_ROUND_UP(...)`, then write.

Reply:
Done.